File: src/client/FantomasTools/ASTViewer/Encoders.fs
module FantomasTools.Client.ASTViewer.Encoders

open ASTViewer.Shared
open FantomasTools.Client.ASTViewer.Model
open Thoth.Json

let encodeUrlModel code model: JsonValue =
    Encode.object
        [ "defines", Encode.string model.Defines
          "isFsi", Encode.bool model.IsFsi
          "code", Encode.string code ]

let encodeInput (input:Input) =
    Encode.object
        [ "sourceCode", Encode.string input.SourceCode
          "defines", (Array.map Encode.string input.Defines |> Encode.array)
          "isFsi", Encode.bool input.IsFsi ]
    |> Encode.toString 2

File: _dist_/bin/ASTViewer/Encoders.js
import { toString, object } from "../.fable/Thoth.Json.5.0.0/Encode.fs.js";
import { map } from "../.fable/fable-library.3.1.7/Array.js";

export function encodeUrlModel(code, model) {
    return object([["defines", model.Defines], ["isFsi", model.IsFsi], ["code", code]]);
}

export function encodeInput(input) {
    return toString(2, object([["sourceCode", input.SourceCode], ["defines", map((value_1) => value_1, input.Defines)], ["isFsi", input.IsFsi]]));
}

